pkg/server/plugin/upstreamca/spire/spire_workload.go

```diff
@@ -0,0 +1,131 @@
+package spireplugin
+
+import (
+	"context"
+	"crypto/x509"
+	"time"
+	"net"
+
+	proto "github.com/spiffe/spire/proto/api/workload"
+	"github.com/spiffe/spire/api/workload"
+	"encoding/pem"
+	"io/ioutil"
+	"os"
+)
+
+//func (m *spirePlugin) getWorkloadSVID(ctx context.Context, config *Configuration) ([]byte, []byte , []byte, string, error) {
+func (m *spirePlugin) getWorkloadSVID(ctx context.Context, config *Configuration) (error) {
+	errorChan := make(chan error, 1)
+
+	duration, _ := time.ParseDuration(config.TTL)
+	wapiClient := m.newWorkloadAPIClient(config.ServerAgentAddr, duration)
+	updateChan := wapiClient.UpdateChan()
+	go func() {
+		err := wapiClient.Start()
+		if err != nil {
+			errorChan <- err
+		}
+	}()
+
+	defer wapiClient.Stop()
+
+	for {
+		select {
+		case svidResponse := <- updateChan:
+			//workloadCert, workloadKey,workloadBundle,spiffeid := m.receiveUpdatedCerts(svidResponse)
+			workloadCert, workloadKey,workloadBundle, spiffeID := m.receiveUpdatedCerts(svidResponse)
+			m.debugDumpCerts(workloadCert, workloadKey, workloadBundle)
+			m.workloadCert = make([]byte, len(workloadCert))
+			copy(m.workloadCert,workloadCert)
+			m.workloadKey = make([]byte, len(workloadKey))
+			copy(m.workloadKey,workloadKey)
+			m.workloadBundle = make([]byte, len(workloadBundle))
+			copy(m.workloadBundle,workloadBundle)
+			m.spiffeID = spiffeID
```

Comment:
Check out this PR I just opened, I think it might make these bits a little easier to deal with

https://github.com/spiffe/spire/pull/649

Reply:
Removed this code and used the direct return from response structure .